sqlserver/datadog_checks/sqlserver/statements.py

```diff
@@ -48,31 +49,33 @@
     "total_spills",
 ]
 
+
 STATEMENT_METRICS_QUERY = """\
 with qstats as (
-    select TOP {limit} query_hash, query_plan_hash, last_execution_time,
+    select query_hash, query_plan_hash, last_execution_time,
```

Comment:
ooc what would be required to add a unit test that checks correctness for this statement?

Reply:
I think the existing unit tests provide enough coverage? Unless we want to add larger integration type test where we run the check many times over the course of several minutes against a consistent load to verify the numbers. We did go through that exact exercise as part of the incident to make the sure the change was correct.

Reply:
Which TOP (10000)? There is no ORDER BY so I'm curious what ordering you _expect_. On a busy system, each iterative sample could theoretically pull a completely different set of 10000 rows.

Reply:
hi @AaronBertrand its a good point, and something we considered early on when developing this. Adding an ORDER BY here would considerably reduce performance of the query, but we might consider it in the future as we continually look to improve the overall performance of the check. 

What were you thinking we would order _by_ in this case? most of the columns are monotonic counts so .. perhaps `last_execution_time`?

Reply:
last_execution_time desc maybe, or execution_count desc, it's just odd to me to see top without order by since that means the results are truly nondeterministic.

Reply:
yeah making it configurable is an interesting idea with the default being a non-deterministic TOP, which provides the best performance outcome. My only concern is people shooting themselves in the foot by accident, but we could probably build in safe guards around the query taking too long, etc. I'll ask the rest of the team what they thing + submit a feature request to our board. Thanks @AaronBertrand!